integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -0,0 +1,332 @@
+                  expect.objectContaining({
+                    price: {
+                      id: expect.any(String),
+                      is_calculated_price_price_list: false,
+                      calculated_amount: 3000,
+                      is_original_price_price_list: false,
+                      original_amount: 3000,
+                      currency_code: "usd",
+                      calculated_price: {
+                        id: expect.any(String),
+                        price_list_id: null,
+                        price_list_type: null,
+                        min_quantity: null,
+                        max_quantity: null,
+                      },
+                      original_price: {
+                        id: expect.any(String),
+                        price_list_id: null,
+                        price_list_type: null,
+                        min_quantity: null,
+                        max_quantity: null,
+                      },
+                    },
+                  }),
+                ],
+              }),
+              expect.objectContaining({
+                id: product2.id,
+                variants: [
+                  expect.objectContaining({
+                    price: null,
+                  }),
+                ],
+              }),
+              expect.objectContaining({
+                id: product3.id,
+              }),
+            ])
+          )
+        })
+      })
+
+      describe("GET /store/products/:id", () => {
+        beforeEach(async () => {
+          ;[product, [variant]] = await createProductsWithVariants(
+            productModule,
+            {
+              title: "test product 1",
+              status: ProductStatus.PUBLISHED,
+              variants: [{ title: "test variant 1" }],
+            }
+          )
+        })
+
+        it("should retrieve product successfully", async () => {
+          let response = await api.get(`/store/products/${product.id}`)
+
+          expect(response.status).toEqual(200)
+          expect(response.data.product).toEqual(
+            expect.objectContaining({
+              id: product.id,
+              variants: [
+                expect.objectContaining({
+                  id: expect.any(String),
+                }),
+              ],
+            })
+          )
+        })
+
+        it("should throw error when calculating prices without context", async () => {
+          let error = await api
+            .get(`/store/products/${product.id}?fields=*variants.prices`)
+            .catch((e) => e)
+
+          expect(error.response.status).toEqual(400)
+          expect(error.response.data).toEqual({
+            message:
+              "Pricing parameters (currency_code or region_id) are required to calculate prices",
+            type: "invalid_data",
+          })
+        })
+
+        it("should get product with prices when context is present", async () => {
+          await createVariantPriceSet({
```

Comment:
Same with these calls, if you use the HTTP endpoints you won't need to do this manually

Reply:
will give it a shot 💪🏻